rosidl_generator_c/resource/msg__functions.h.template

```diff
@@ -86,6 +86,66 @@ ROSIDL_GENERATOR_C_PUBLIC_@(spec.base_type.pkg_name)
 void
 @(msg_typename)__destroy(@(msg_typename) * msg);
 
+@# If the message contains an unbounded non-primitive array subfield, declare a resize function
+@{
+unbounded_array_fields = []
+for field in spec.fields:
+    if not field.type.is_array:
+        continue
+    if field.type.is_fixed_size_array():
+        continue
+#    if field.type.is_primitive_type():
+#        continue
+
+    unbounded_array_fields.append(field)
+}@
+
+@[for field in unbounded_array_fields]
+@{
+if field.type.is_primitive_type():
+    subfield_pkg_name = 'rosidl_generator_c'
+    if field.type.type == 'string':
+        subfield_type_name = 'String'
+    else:
+        subfield_type_name = field.type.type
+
+    subfield_array_typename = '%s__%s__Array' % (subfield_pkg_name, subfield_type_name)
+else:
+    subfield_pkg_name = field.type.pkg_name
+    subfield_type_name = field.type.type
+    subfield_array_typename = '%s__msg__%s__Array' % (subfield_pkg_name, subfield_type_name)
```

Comment:
The Python function `msg_type_to_c` already converts a field type into the C type. I think this should be used here instead.